client/src/store/slices/orderSlice/orderSlice.js

```diff
@@ -0,0 +1,63 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import postNewOrder from '../../../api/postNewOrder';
+// import axios from 'axios';
+
+const initialState = {
+  currentStep: 0,
+  isModalOpen: false,
+  isLoading: false,
+  orderInfo: {},
+};
+
+export const createNewOrder = createAsyncThunk('order/createNewOrder', async (orderData, { rejectWithValue }) => {
+  try {
+    const orderInfo = await postNewOrder(orderData);
+
+    // checks if order does not contain order object (possible in cases when server returns response with incorrect data passed in input fields)
+    if (!orderInfo.order) {
+      await Promise.reject(orderInfo);
+    }
+
+    return orderInfo;
+  } catch (e) {
+    return rejectWithValue(e);
+  }
+});
+
+const orderSlice = createSlice({
+  name: 'order',
+  initialState,
+
+  reducers: {
+    increaseStep: (state) => {
+      state.currentStep += 1;
+    },
+    decreaseStep: (state) => {
+      if (state.currentStep === 0) {
+        state.currentStep = 0;
+      } else {
+        state.currentStep -= 1;
+      }
+    },
```

Comment:
```suggestion
    decreaseStep: (state) => {
      if (state.currentStep >= 1) {
        state.currentStep -= 1;
      } 
    },
```

perhaps this approach would be better

Reply:
done